### PR TITLE
In ListComboBox prevents caret from moving on key up/down pressed

### PR DESCRIPTION
**Pull request by hamen**
Relates to #691 

Recording2024-11-18114501-ezgif com-video-to-gif-converter

### CODE REVIEW

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt [hunk 1]
```diff
@@ -247,32 +247,40 @@ private fun TextInput(
...
-                        if (popupExpanded) {
-                            onArrowDownPress()
-                        } else {
-                            onSetPopupExpanded(true)
-                            textFieldFocusRequester.requestFocus()
+                    // We are consuming the event to prevent the caret
+                    // from moving to the start and to the end of the text field
+                    // when the user presses the arrow up/down keys
+                    when {
+                        it.type == KeyEventType.KeyDown && it.key == Key.DirectionDown -> {
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt:254]: Nit: you can extract the type check to an if that quick-returns false if it's not KeyDown, before the when, simplifying the when conditions (which all check that)

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt [hunk 2]
```diff
@@ -247,32 +247,40 @@ private fun TextInput(
...
-                    }
-
-                    if (it.type == KeyEventType.KeyDown && it.key == Key.Enter) {
-                        if (popupExpanded) {
+                        it.type == KeyEventType.KeyDown && it.key == Key.Enter -> {
+                            if (popupExpanded) {
+                                onSetPopupExpanded(false)
+                            }
+                            onEnterPress()
+                            false
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt:276]: I know it was already essentially doing this, but shouldn't this be `true` given we're consuming the enter press?

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt [hunk 3]
```diff
@@ -247,32 +247,40 @@ private fun TextInput(
...
+                        }
+                        it.type == KeyEventType.KeyDown && it.key == Key.Escape && popupExpanded -> {
                             onSetPopupExpanded(false)
+                            false
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt:280]: Ditto

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt [hunk 4]
```diff
@@ -247,32 +247,40 @@ private fun TextInput(
...
                 .onPreviewKeyEvent {
-                    if (it.type == KeyEventType.KeyDown && it.key == Key.DirectionDown) {
-                        if (popupExpanded) {
-                            onArrowDownPress()
-                        } else {
-                            onSetPopupExpanded(true)
-                            textFieldFocusRequester.requestFocus()
+                    // We are consuming the event to prevent the caret
+                    // from moving to the start and to the end of the text field
+                    // when the user presses the arrow up/down keys
```
Comment (rock3r) [inline ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt:252]: This comment here was misleading for me — it's related to L261 and L269 only, but placed far from either; also, other branches in the when should return `true` as well...

Comment (hamen) [reply ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt]: 💯 Removed in https://github.com/JetBrains/jewel/pull/692/commits/05adbe92c586e327936c5d86a8021d3747471685

File: ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt (hunk 1 above)
Comment (hamen) [reply ui/src/main/kotlin/org/jetbrains/jewel/ui/component/EditableComboBox.kt]: Done in https://github.com/JetBrains/jewel/pull/692/commits/05adbe92c586e327936c5d86a8021d3747471685

**Pull request merged**